src/hotspot/share/gc/g1/g1FullGCMarker.inline.hpp

```diff
@@ -54,12 +54,10 @@ inline bool G1FullGCMarker::mark_object(oop obj) {
   }
 
   // Marked by us, preserve if needed.
-  markWord mark = obj->mark();
-  if (obj->mark_must_be_preserved(mark) &&
-      // It is not necessary to preserve marks for objects in regions we do not
-      // compact because we do not change their headers (i.e. forward them).
-      _collector->is_compacting(obj)) {
-    preserved_stack()->push(obj, mark);
+  if (_collector->is_compacting(obj)) {
```

Comment:
I wonder if is_compacting() is actually used (returning false) by any collector. Sounds like it might be a left-over from CMS.

Reply:
`is_compacting()` returns info in `G1FullCollector::_region_attr_table`. (If I put `assert(false)` inside this `if`, I get an assert failure.)